include/seqan3/align/alignment.hpp

```diff
@@ -0,0 +1,196 @@
+//
+// Copyright (c) 2006-2017, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2017, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+// Author: Joerg Winkler <j.winkler AT fu-berlin.de>
+// ============================================================================
+
+#pragma once
+
+#include <iomanip>
+#include <iostream>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+#include <range/v3/algorithm/for_each.hpp>
+#include <range/v3/iterator_range.hpp>
+#include <range/v3/view/all.hpp>
+#include <range/v3/view/slice.hpp>
+#include <range/v3/view/zip.hpp>
+
+#include <seqan3/alphabet.hpp>
+#include <seqan3/range/view/to_char.hpp>
+
+namespace seqan3
+{
+
+template <typename t>
+concept bool aligned_sequence_concept = true;
+
+//! Alignment class.
+/*!
+ * An alignment is a tuple of at least two aligned sequences.
+ * \tparam sequence_ts sequence types of the alignment
+ */
+template <typename ...sequence_ts>
+    requires (aligned_sequence_concept<sequence_ts> && ...)
+class alignment : public std::tuple<sequence_ts...>
+{
+public:
+    //! The alignment depth is the number of sequences contained.
+    static std::size_t const depth = sizeof...(sequence_ts);
+
+    //! Constructor that allows to pass sequences directly.
+    constexpr alignment(sequence_ts && ...sequences)
```

Comment:
As I read the code, only a list of sequences is allowed for construction, but what about a container of sequences? There should maybe also be a function to add a sequence to an existing align object IMPOV. What do you think ?

Reply:
As an alignment is a tuple, the only way to add a sequence would be `std::tuple_cat` and return a new object that is the concatenation the current alignment with the new sequence. I can write a global function – do we want that?

```cpp
template <typename sequence_type, typename... alignment_type>
alignment<alignment_type..., sequence_type> append_sequence(alignment<alignment_type...> const & ali, sequence_type const & new_seq)
{
    return std::tuple_cat(ali, std::make_tuple(new_seq));
}
```

Reply:
It's not possible, but also in most use cases you know the dimension, or not?